lib/Sema/TypeCheckDecl.cpp

```diff
@@ -478,6 +478,52 @@ static void checkInheritanceClause(
   }
 }
 
+// Check for static properties that produce empty option sets
+static void checkForEmptyOptionSet(const VarDecl *VD, TypeChecker &tc) {
+  if (!VD->isStatic() || !VD->isLet())
+    return;
+  
+  auto DC = VD->getDeclContext();
+  
+  auto *optionSetProto = tc.Context.getProtocol(KnownProtocolKind::OptionSet);
+  auto protocolConformance = tc.containsProtocol(DC->getSelfTypeInContext(), optionSetProto, DC, /*Flags*/None);
+  if (!protocolConformance)
+    return;
+  
+  auto type = VD->getType();
+  if (!type->isEqual(DC->getSelfTypeInContext()))
+    return;
+  
+  auto PBD = VD->getParentPatternBinding();
+  if (!PBD)
+    return;
+  
+  for (auto entry : PBD->getPatternList()) {
+    if (entry.getPattern()->getSingleVar() != VD) continue;
+    
+    auto ctor = dyn_cast<CallExpr>(entry.getInit());
+    if (!ctor) continue;
+    if (!isa<ConstructorDecl>(ctor->getCalledValue())) continue;
+    
+    if (ctor->getNumArguments() != 1) continue;
+    auto argLabels = ctor->getArgumentLabels();
+    if (argLabels.front() != tc.Context.Id_rawValue) continue;
+    
+    auto *args = cast<TupleExpr>(ctor->getArg());
+    auto intArg = dyn_cast<IntegerLiteralExpr>(args->getElement(0));
+    if (!intArg) continue;
+    
+    auto val = intArg->getValue();
+    if (val != 0) continue;
```

Comment:
Some of these can be inlined as they're not being used later. 

```suggestion
    if (intArg->getValue() != 0) continue;
```